api/types/accesslist/convert/v1/accesslist.go

```diff
@@ -33,95 +33,34 @@ type AccessListOption func(*accesslist.AccessList)
 
 // FromProto converts a v1 access list into an internal access list object.
 func FromProto(msg *accesslistv1.AccessList, opts ...AccessListOption) (*accesslist.AccessList, error) {
-	if msg == nil {
-		return nil, trace.BadParameter("access list message is nil")
-	}
-
-	if msg.Spec == nil {
+	spec := msg.GetSpec()
+	if spec == nil {
 		return nil, trace.BadParameter("spec is missing")
 	}
-	if msg.Spec.Audit == nil {
-		return nil, trace.BadParameter("audit is missing")
-	}
-	if msg.Spec.MembershipRequires == nil {
-		return nil, trace.BadParameter("membershipRequires is missing")
-	}
-	if msg.Spec.OwnershipRequires == nil {
-		return nil, trace.BadParameter("ownershipRequires is missing")
-	}
-	if msg.Spec.Grants == nil {
+	if spec.GetGrants() == nil {
 		return nil, trace.BadParameter("grants is missing")
 	}
 
-	var recurrence accesslist.Recurrence
-	if msg.Spec.Audit.Recurrence != nil {
-		recurrence.Frequency = accesslist.ReviewFrequency(msg.Spec.Audit.Recurrence.Frequency)
-		recurrence.DayOfMonth = accesslist.ReviewDayOfMonth(msg.Spec.Audit.Recurrence.DayOfMonth)
-	}
-
-	var notifications accesslist.Notifications
-	if msg.Spec.Audit.Notifications != nil {
-		if msg.Spec.Audit.Notifications.Start != nil {
-			notifications.Start = msg.Spec.Audit.Notifications.Start.AsDuration()
-		}
-	}
-
-	owners := make([]accesslist.Owner, len(msg.Spec.Owners))
-	for i, owner := range msg.Spec.Owners {
-		owners[i] = FromOwnerProto(owner)
-		// Set IneligibleStatus to empty as default.
-		// Must provide as options to set it with the provided value.
-		owners[i].IneligibleStatus = ""
-	}
-
-	var ownerGrants accesslist.Grants
-	if msg.Spec.OwnerGrants != nil {
-		ownerGrants.Roles = msg.Spec.OwnerGrants.Roles
-		if msg.Spec.OwnerGrants.Traits != nil {
-			ownerGrants.Traits = traitv1.FromProto(msg.Spec.OwnerGrants.Traits)
-		}
-	}
+	metadata := headerv1.FromMetadataProto(msg.GetHeader().GetMetadata())
```

Comment:
Do we need to verify `GetMetadata` doesn't return nil?

Reply:
No, The convert function should not validate the data and not preform any domain validation. 

For that we have accesslist.CheckAndSetDefault where the Metadata are validated.  https://github.com/gravitational/teleport/blob/mastser/api/types/accesslist/member.go#L89